recipes/libtins/all/conanfile.py

```diff
@@ -0,0 +1,102 @@
+from conans import tools, CMake, ConanFile
+import os
+
+
+class LibTinsConan(ConanFile):
+    name = "libtins"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mfontanini/libtins"
+    description = "High-level, multiplatform C++ network packet sniffing and crafting library."
+    license = "BSD-2-Clause"
+    topics = ("pcap", "packets", "network", "packet-analyser", "packet-parsing", "libpcap", "sniffing")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ack_tracker": [True, False],
+        "with_wpa2": [True, False],
+        "with_dot11": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ack_tracker": True,
+        "with_wpa2": True,
+        "with_dot11": True,
+    }
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libpcap/1.10.0")
+        if self.options.with_ack_tracker:
+            self.requires("boost/1.75.0")
+        if self.options.with_wpa2:
+            self.requires("openssl/1.1.1i")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        # Workaround for casing issues in cmake_find_package generator of openssl recipe
+        top_cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        src_cmakelists = os.path.join(self._source_subfolder, "src", "CMakeLists.txt")
+        tools.replace_in_file(top_cmakelists, "OPENSSL_FOUND", "OpenSSL_FOUND")
+        tools.replace_in_file(src_cmakelists, "OPENSSL_INCLUDE_DIR", "OpenSSL_INCLUDE_DIR")
+        tools.replace_in_file(src_cmakelists, "OPENSSL_LIBRARIES", "OpenSSL_LIBRARIES")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LIBTINS_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["LIBTINS_BUILD_TESTS"] = False
+
+        self._cmake.definitions["LIBTINS_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["LIBTINS_ENABLE_CXX11"] = tools.valid_min_cppstd(self, 11)
```

Comment:
This is never clear in my head, does this take into account the compiler's default and the build settings?

Reply:
it takes into account compiler.cppstd, and if not set default cpp standard of compiler.